tests/basic/vcluster_test.go

```diff
@@ -430,8 +432,460 @@ var _ = Describe("{VolumeDriverDownVCluster}", func() {
+			}
+		})
+		// Validate if Nginx Deployment is healthy or not
+		err = vc.IsDeploymentHealthy(appNS, deploymentName, 1)
+		log.FailOnError(err, "Looks like Nginx Deployment is not healthy")
+		log.Infof("Nginx Deployment %s is healthy. Will kill Px and wait for its restart on all nodes now", deploymentName)
+	})
+	JustAfterEach(func() {
+		// VCluster, StorageClass and Namespace cleanup
+		err := vc.VClusterCleanup(scName)
+		if err != nil {
+			log.Errorf("Problem in Cleanup: %v", err)
+		} else {
+			log.Infof("Cleanup successfully done.")
+		}
+	})
+})
+
+// This Test Case is failing due to : https://portworx.atlassian.net/browse/PWX-34792
+var _ = Describe("{VolumeSnapshotAndRestoreVcluster}", func() {
+	vc := &vcluster.VCluster{}
+	var scName string
+	var pvcName string
+	var appNS string
+	do_verify := 1
+	verify := "crc32c"
+	fioOptions := vcluster.FIOOptions{
+		Name:      "mytest",
+		IOEngine:  "libaio",
+		RW:        "write",
+		BS:        "4k",
+		NumJobs:   1,
+		Size:      "500m",
+		TimeBased: true,
+		Runtime:   "100s",
+		Filename:  "/data/fiotest",
+		EndFsync:  1,
+		DoVerify:  &do_verify,
+		Verify:    &verify,
+	}
+	JustBeforeEach(func() {
+		StartTorpedoTest("VolumeSnapshotAndRestoreVcluster", "Create, Connect and run FIO Application on Vcluster, Create a Volume Snapshot, Restore it and Read from FIO the data previously written", nil, 0)
+		vc, err = vcluster.NewVCluster("my-vcluster1")
+		log.FailOnError(err, "Failed to initialise VCluster")
+		err = vc.CreateAndWaitVCluster()
+		log.FailOnError(err, "Failed to create VCluster")
+	})
+	It("Create, Connect and run FIO Application on Vcluster, Create a Volume Snapshot, Restore it and Read from FIO the data previously written", func() {
+		// Create Snapshot Schedule Policy
+		snapSchedulePolicy := fmt.Sprintf("snap-schedule-%v", time.Now().Unix())
+		err := snapshotutils.SchedulePolicyInDefaultNamespace(snapSchedulePolicy, 1, 5)
+		log.FailOnError(err, "Failed to create Snapshot Schedule Policy")
+		// Create Storage Class with snapshot schedule policy
+		scName = fmt.Sprintf("fio-app-sc-%v", time.Now().Unix())
+		err = CreateStorageClass(scName, WithSnapshotSchedule(snapSchedulePolicy, "local"))
+		log.FailOnError(err, "Error creating Storageclass")
+		log.Infof("Successfully created StorageClass with name: %v", scName)
+		// Create PVC on VCluster
+		appNS = scName + "-ns"
+		pvcName, err = vc.CreatePVC("", scName, appNS, "")
+		log.FailOnError(err, fmt.Sprintf("Error creating PVC with Storageclass name %v", scName))
+		log.Infof("Successfully created PVC with name: %v", pvcName)
+		jobName := "fio-job"
+		// Create FIO Deployment on VCluster using the above PVC
+		err = vc.CreateFIODeployment(pvcName, appNS, fioOptions, jobName)
+		log.FailOnError(err, "Error in creating FIO Application")
+		log.Infof("Successfully ran FIO on Vcluster")
+		log.Infof("Waiting for 60 seconds as that is frequency to take one snapshot")
+		time.Sleep(1 * time.Minute)
```

Comment:
Using sleep could have false positive result in some cases where it look few ms/second extra. Better to make sure snaplist should not be empty..

Reply:
Actually that won't ever happen in this scenario as we are tracking the latest snapshot. So even if we are late, it will simply take the last latest snapshot which happened before this sleep 2-3 times.